rssfeed/tasks.py

```diff
@@ -0,0 +1,171 @@
+from datetime import datetime
+from time import mktime
+
+import feedparser
+import BeautifulSoup
+from celery.schedules import crontab
+from django.utils import timezone
+from celery.task import periodic_task
+from rssfeed.models import Entry, Feed
+
+MAX = 20
+MAX_LENGTH = 2000
+
+
+@periodic_task(
+    run_every=crontab(
+        hour='*',
+        minute='*/15',
+        day_of_week='*'),
+    ignore_result=True
+)
+def poll_feed(pk_feed, verbose=False):
+    """
+    Read through a feed looking for new entries.
+
+    db_feed: The database field containing the feed in question
+    verbose: True for debugging purposes
+
+    """
+    db_feed = Feed.objects.get(pk=pk_feed)
+    parsed = feedparser.parse(db_feed.url)
+
+    if hasattr(parsed.feed, "bozo_exception"):
+        if verbose:
+            # Malformed feed
+            msg = 'Rssfeed poll_feeds found Malformed feed, "%s": %s' % (
+                db_feed.url, parsed.feed.bozo_exception)
+            print(msg)
+        return
+
+    if hasattr(parsed.feed, "published_parsed"):
+        published = datetime.fromtimestamp(
+            mktime(parsed.feed.published_parsed)
+        )
+        db_feed.published = published
+
+    for attr in ["title", "title_detail", "link"]:
+        if not hasattr(parsed.feed, attr):
+            if verbose:
+                msg = 'rssfeed poll_feeds. Feed "%s" has no %s' % (
+                    db_feed.url, attr)
+                print(msg)
+            return
+
+    # Get the title of the RSS feed
+    if len(parsed.feed.title) > MAX_LENGTH:
+        db_feed.title = parsed.feed.title[0:MAX_LENGTH - 1]
+    else:
+        db_feed.title = parsed.feed.title
+
+    if hasattr(parsed.feed, "description_detail") and hasattr(parsed.feed,
+                                                              "description"):
+        db_feed.description = parsed.feed.description
+    else:
+        db_feed.description = ""
+    db_feed.last_polled = timezone.now()
+
+    if hasattr(parsed.feed, "image"):
+        if len(parsed.feed.image.href) > MAX_LENGTH:
+            db_feed.image = parsed.feed.image.href[0:MAX_LENGTH - 1]
```

Comment:
Hmm, the reason I did this, is because in the event that the href is > 2000 characters, I made sure that it limits it physically. Is there a better way to do this?